code/datums/skills/_skill.dm

```diff
@@ -0,0 +1,105 @@
+#define	SKILL_NONE  0
+#define	SKILL_BASIC  1
+#define	SKILL_ADEPT  2
+#define	SKILL_EXPERT  3
+
+var/global/list/SKILLS = null
+
+/datum/skillsystem/
+	var/datum/skillsystem/preset/presetfriend = /datum/skillsystem/preset
+	
+/datum/skillsystem/skill/
+    var/ID = "none" // ID of the skill, used in code
+    var/name = "None" // name of the skill
+    var/desc = "Placeholder skill" // detailed description of the skill
+    var/field = "Misc" // the field under which the skill will be listed
+    var/secondary = 0 // secondary skills only have two levels and cost significantly less
+
+/datum/attribute/var
+	ID = "none"
+	name = "None"
+	desc = "This is a placeholder"
+
+/datum/skillsystem/proc/CalculateSkillClass(points, age)
+	if(points <= 0) return "Unconfigured"
+	// skill classes describe how your character compares in total points
+	points -= min(round((age - 20) / 2.5), 4) // every 2.5 years after 20, one extra skillpoint
+	if(age > 30)
+		points -= round((age - 30) / 5) // every 5 years after 30, one extra skillpoint
+	switch(points)
+		if(-1000 to 24)
+			return "Reasonable"
+		if(24 to 1000)
+			return "Unreasonable"
+
+/datum/skillsystem/proc/SkillCheck(var/mob/living/carbon/human/M, var/skill, var/level)
+	if(!skill || !level || !istype(M))
+		return 0
+	if(SKILLS == null)
+		gskillsystem.setup_skills()
+	if(!M.skills || M.skills.len == 0)
+		return 0
+	for(var/V in SKILLS)
+		for(var/datum/skillsystem/skill/S in SKILLS[V])
+			if(!(S.name == skill))
+				continue
+			if(M.skills[S.ID] >= level)
+				return 1
+	return 0
+
+/datum/skillsystem/proc/setup_skills()
+	if(SKILLS == null)
+		SKILLS = list()
+		for(var/T in (typesof(/datum/skillsystem/skill/)-/datum/skillsystem/skill))
+			var/datum/skillsystem/skill/S = new T
+			if(S.ID != "none")
+				if(!SKILLS.Find(S.field))
```

Comment:
SKILLS is assoc here, so you could replace this if and its body with just `LAZYINITLIST(SKILLS[S.field])`.